codegen/src/main/ops/org/nd4j/codegen/ops/Linalg.kt

```diff
@@ -0,0 +1,112 @@
+        Output(DataType.NDARRAY, "output"){ description = "Transformed tensor" }
+
+        Doc(Language.ANY, DocScope.ALL){
+            """
+             Computes the Cholesky decomposition of one or more square matrices.
+            """.trimIndent()
+        }
+    }
+
+    Op("Lstsq") {
+        javaPackage = "org.nd4j.linalg.api.ops.custom"
+        javaOpClass = "Lstsq"
+
+        Input(DataType.NDARRAY, "matrix") {description = "input tensor"}
+        Input(DataType.NDARRAY, "rhs") {description = "input tensor"}
+        Arg(DataType.FLOATING_POINT, "l2_reguralizer") {description = "regularizer"}
+        Arg(DataType.BOOL, "fast") {description = "fast mode, defaults to True"}
+        Output(DataType.NDARRAY, "output"){ description = "Transformed tensor" }
+
+        Doc(Language.ANY, DocScope.ALL){
+            """
+             Solver for linear squares problems.
+            """.trimIndent()
+        }
+    }
+
+    Op("Solve") {
+        javaPackage = "org.nd4j.linalg.api.ops.custom"
+        javaOpClass = "LinearSolve"
+
+        Input(DataType.NDARRAY, "matrix") {description = "input tensor"}
+        Input(DataType.NDARRAY, "rhs") {description = "input tensor"}
+        Arg(DataType.BOOL, "adjoint") {description = "adjoint mode, defaults to False"}
+        Output(DataType.NDARRAY, "output"){ description = "Output tensor" }
+
+        Doc(Language.ANY, DocScope.ALL){
+            """
+             Solver for systems of linear questions.
+            """.trimIndent()
+        }
+    }
+
+    Op("TriangularSolve") {
+        javaPackage = "org.nd4j.linalg.api.ops.custom"
+        javaOpClass = "TriangularSolve"
+
+        Input(DataType.NDARRAY, "matrix") {description = "input tensor"}
+        Input(DataType.NDARRAY, "rhs") {description = "input tensor"}
+        Arg(DataType.BOOL, "lower") {description = "defines whether innermost matrices in matrix are lower or upper triangular"}
+        Arg(DataType.BOOL, "adjoint") {description = "adjoint mode"}
+
+        Doc(Language.ANY, DocScope.ALL){
+            """
+             Solver for systems of linear questions.
+            """.trimIndent()
+        }
+    }
+
+    Op("Lu") {
+        javaPackage = "org.nd4j.linalg.api.ops.custom"
+        javaOpClass = "Lu"
+
+        Input(DataType.NDARRAY, "input") {description = "input tensor"}
+
+        Doc(Language.ANY, DocScope.ALL){
+            """
+             Computes LU decomposition.
+            """.trimIndent()
+        }
+    }
+
+    Op("Matmul") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.transforms.pairwise"
+        javaOpClass = "MatMul"
+
+        Input(DataType.NDARRAY, "a") {description = "input tensor"}
+        Input(DataType.NDARRAY, "b") {description = "input tensor"}
+
+        Doc(Language.ANY, DocScope.ALL){
+            """
+             Performs matrix mutiplication on input tensors.
+            """.trimIndent()
+        }
+    }
```

Comment:
TriangularSolve, Lu and Matmul all need output array definitions to be defined.
And input datatypes need to be correct types (float, numerical, whatever), not just NDArray

We've already got mmul defined in SDBaseOps; we probably want to add a reference to that instead of having code duplication. Not sure on the best way to do that yet though...

Reply:
Fixed.